packages/core/js-client/src/clientPeer/__test__/client.spec.ts

```diff
@@ -26,162 +45,179 @@ describe('FluenceClient usage test suite', () => {
         )
     )`;
 
-            const particle = await peer.internals.createNewParticle(script);
-            
-            const result = await new Promise<string>((resolve, reject) => {
-                if (particle instanceof Error) {
-                    return reject(particle.message);
-                }
-
-                registerHandlersHelper(peer, particle, {
-                    load: {
-                        relay: () => {
-                            return peer.getRelayPeerId();
-                        },
-                    },
-                    callback: {
-                        callback: (args: any) => {
-                            const [val] = args;
-                            resolve(val);
-                        },
-                        error: (args: any) => {
-                            const [error] = args;
-                            reject(error);
-                        },
-                    },
-                });
-
-                peer.internals.initiateParticle(particle, handleTimeout(reject));
-            });
-
-            expect(result).toBe('hello world!');
+      const particle = await peer.internals.createNewParticle(script);
+
+      const result = await new Promise<JSONValue>((resolve, reject) => {
+        if (particle instanceof Error) {
+          reject(particle.message);
+          return;
+        }
+
+        registerHandlersHelper(peer, particle, {
+          load: {
+            relay: () => {
+              return peer.getRelayPeerId();
+            },
+          },
+          callback: {
+            callback: (args): undefined => {
```

Comment:
I would just allow `void` return type at `registerHandlersHelper` so you don't need to do this

Reply:
got it